### ISSUE TITLE
test_codonalign.Test_dn_ds regression on scipy 1.5.0

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#2993]
TravisCI Python 3.7 has started failing on the master branch:

```
certifi==2020.6.20,chardet==3.0.4,codecov==2.1.7,coverage==5.1,cycler==0.10.0,idna==2.9,kiwisolver==1.2.0,matplotlib==3.2.2,mmtf-python==1.1.2,msgpack==1.0.0,mysql-connector-python-rf==2.2.2,mysqlclient==1.4.6,numpy==1.19.0,psycopg2-binary==2.8.5,pyparsing==2.4.7,python-dateutil==2.8.1,requests==2.24.0,scipy==1.5.0,six==1.15.0,urllib3==1.25.9
...
======================================================================
FAIL: test_dn_ds (test_codonalign.Test_dn_ds)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_codonalign.py", line 250, in test_dn_ds
    self.assertAlmostEqual(dN, 0.0194, places=4)
AssertionError: 0.018551097874995984 != 0.0194 within 4 places (0.00084890212500401616 difference)
----------------------------------------------------------------------
```

See https://travis-ci.org/github/biopython/biopython/jobs/700971840

Previous build worked, but had scipy 1.4.1

```
certifi==2020.6.20,chardet==3.0.4,codecov==2.1.7,coverage==5.1,cycler==0.10.0,idna==2.9,kiwisolver==1.2.0,matplotlib==3.2.2,mmtf-python==1.1.2,msgpack==1.0.0,mysql-connector-python-rf==2.2.2,mysqlclient==1.4.6,numpy==1.19.0,psycopg2-binary==2.8.5,pyparsing==2.4.7,python-dateutil==2.8.1,requests==2.24.0,scipy==1.4.1,six==1.15.0,urllib3==1.25.9
```

See https://travis-ci.org/github/biopython/biopython/jobs/700563827

This is currently the only TravisCI target which has SciPy installed, and thus only Python 3.7 shows the problem.

**Comment by peterjc** [comment]
Failure is here:

```python
        try:
            # New in scipy v0.11
            from scipy.optimize import minimize
            dN, dS = cal_dn_ds(codon_seq1, codon_seq2, method="ML")
            self.assertAlmostEqual(dN, 0.0194, places=4)
            self.assertAlmostEqual(dS, 0.0217, places=4)
        except ImportError:
            # TODO - Show a warning?
            pass
```

https://github.com/biopython/biopython/blob/biopython-177/Tests/test_codonalign.py#L250

Most likely a chance in ``scipy.optimize.minimize``?

**Comment by peterjc** [comment]
This test ends up calling the ``_ml`` function:

https://github.com/biopython/biopython/blob/biopython-177/Bio/codonalign/codonseq.py#L1130

This calls ``scipy.optimize.minimize`` as follows:

```
    opt_res = minimize(
        func,
        [1, 0.1, 2],
        method="L-BFGS-B",
        bounds=((1e-10, 20), (1e-10, 20), (1e-10, 10)),
        tol=1e-5,
    )
```

https://docs.scipy.org/doc/scipy/reference/release.1.5.0.html says:

> If the minimize method uses bounds the numerical differentiation strictly obeys those limits.

**Comment by peterjc** [comment]
Short term workaround applied so TravisCI continues to pass (#2994)

**Comment by peterjc** [comment]
@zruan Are you able to look at this change to the SciPy minimiser as used in for DN/DS calculation?

**Comment by peterjc** [comment]
Looks like this is breaking on AppVeyor too now...

**Comment by peterjc** [comment]
Closed by #2998.
